src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Download/PackageDownloadRunner.cs

```diff
@@ -271,16 +338,18 @@ private static bool DetectAndReportInsecureSources(
             return false;
         }
 
-        private static IReadOnlyList<SourceRepository> GetSourceRepositories(IReadOnlyList<PackageSource> packageSources)
+        private static (IReadOnlyDictionary<string, SourceRepository>, List<SourceRepository>) GetSourceRepositories(IReadOnlyList<PackageSource> packageSources)
         {
             IEnumerable<Lazy<INuGetResourceProvider>> providers = Repository.Provider.GetCoreV3();
-            List<SourceRepository> sourceRepositories = [];
+            Dictionary<string, SourceRepository> sourceRepositories = [];
```

Comment:
perf is not super critical here, but a good idea to get into the habbit of providing the size when we know it.